container/rules/debian_pkg_tar.bzl

```diff
@@ -0,0 +1,185 @@
+    "image_tar": attr.label(
+        allow_files = True,
+        single_file = True,
+    ),
+    "image": attr.label(
+        allow_files = True,
+        single_file = True,
+    ),
+    "packages": attr.string_list(),
+    "keys": attr.label_list(
+        allow_files = True,
+    ),
+}
+
+InstallableTarInfo = provider(fields = [
+    "installables_tar",
+])
+
+def _generate_deb_tar(
+        ctx,
+        packages = None,
+        additional_repos = None,
+        keys = None,
+        download_pkgs_output_tar = None,
+        download_pkgs_output_script = None):
+    """A function for producing a tarball.
+
+    _generate_deb_tar function produces a tarball with all debian packages, additional_repos, and keys.
+
+    Args:
+      ctx: ctx as the same as for container_image + list of language_tool_layer(s)
+      https://github.com/bazelbuild/rules_docker#container_image
+      packages: packages aggregated from each language_tool_layer by toolchain_container rule
+      additional_repos: additional_repos aggregated from each language_tool_layer by
+      toolchain_container rule
+      keys: keys aggregated from each language_tool_layer by toolchain_container rule
+      download_pkgs_output_tar: output tar file generated by download_pkgs rule to
+      override default output_tar name
+      download_pkgs_output_script: output script generated by download_pkgs rule to
+      override default output_script name
+    """
+
+    packages = packages or ctx.attr.packages
+    additional_repos = additional_repos or ctx.attr.additional_repos
+    keys = keys or ctx.attr.keys
+
+    # Prepare base image for the download_pkgs rule.
+    download_base = ctx.files.base[0]
+
+    # Create an intermediate image with additional gpg keys used to download packages.
+    if keys != []:
+        image_with_keys = "%s_with_keys" % ctx.attr.name
+
+        # Declare intermediate output file generated by add_apt_key rule.
+        image_with_keys_output_executable = ctx.actions.declare_file(image_with_keys)
+        image_with_keys_output_tarball = ctx.actions.declare_file(image_with_keys + ".tar")
+        image_with_keys_output_layer = ctx.actions.declare_file(image_with_keys + "-layer.tar")
+
+        _apt_key.implementation(
+            ctx,
+            name = image_with_keys,
+            image_tar = ctx.files.base[0],
+            keys = keys,
+            output_executable = image_with_keys_output_executable,
+            output_tarball = image_with_keys_output_tarball,
+            output_layer = image_with_keys_output_layer,
+        )
+        download_base = image_with_keys_output_tarball
+
+    # Declare intermediate output file generated by download_pkgs rule.
+    download_pkgs_output_executable = ctx.actions.declare_file(ctx.attr.name + "-download_pkgs_output_executable.sh")
+    if download_pkgs_output_tar:
+        download_pkgs_output_tar = ctx.actions.declare_file(download_pkgs_output_tar)
```

Comment:
declare a var and give it value 
download_pkgs_output_tar or ctx.attr.name + ".tar" depending on the value of download_pkgs_output_tar (in a one line if-condition-assignment), then just assign the value to a new variable "output_tar"(do not overload the one passed as parameter. (Same for output_script)

Reply:
Yea, will make such changes, thanks for the suggestion.